### PR TITLE
Feature/custom calender/ap 2395

### PR DESCRIPTION
**Pull request by nolantellis**
* Added Holiday with tests
* Added logic to calculate duration
* This does not effect any development features.

### CODE REVIEW

File: Apromore-Calendar/src/main/java/org/apromore/calendar/service/CustomCalendarService.java [hunk 1]
```diff
@@ -38,68 +40,86 @@
...
+    OffsetTime endTime = OffsetTime.of(LocalTime.of(17, 0), ZoneOffset.UTC);
+
+    CustomCalendar customCalender = createCalendar(description, weekendsOff, startTime, endTime);
+    CalenderModel calenderModel = modelMapper.getMapper().map(customCalender, CalenderModel.class);
+
+    return calenderModel;
+
+  }
+
+  public CalenderModel getCalenderById(Long id) {
```
Comment (dreamind) [inline Apromore-Calendar/src/main/java/org/apromore/calendar/service/CustomCalendarService.java]: Calendar instead of Calender?

**Comment by dreamind** [comment]
While all the workday and holiday definition is set in UTC, should we add the TimeZone to the CalendarModel?
Is the duration calculated in the UTC context? To account for TimeZone specific quirks (like daylight saving) should we use ZonedDateTime (and perhaps ChronoUnit.between)?

**Review by raboczi — CHANGES_REQUESTED** [review]
As already noted by Ivo, CalenderModel needs to be renamed as CalendarModel.  Additionally, you should configure your IDE to indent with spaces rather than tabs; it seems to keep rewriting swathes of XML.

File: Apromore-Calendar/src/main/java/org/apromore/calendar/service/CustomCalendarService.java (hunk 1 above)
Comment (nolantellis) [reply Apromore-Calendar/src/main/java/org/apromore/calendar/service/CustomCalendarService.java]: Corrected.

**Comment by nolantellis** [comment]
> While all the workday and holiday definition is set in UTC, should we add the TimeZone to the CalendarModel?
> Is the duration calculated in the UTC context? To account for TimeZone specific quirks (like daylight saving) should we use ZonedDateTime (and perhaps ChronoUnit.between)?

You are right. I will merge this changes which does not effect anything. I have created another task to convert it to ZonedDateTime

**Review by dreamind — APPROVED** [review]

**Review by raboczi — APPROVED** [review]
Good enough to merge!

**Pull request merged**